### ISSUE TITLE
PHP options should be passed to child processes

### ISSUE THREAD

**Issue by robsontenorio** [sebastianbergmann/phpunit#4664]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 9.5
| PHP version         | 8.0.3
| Installation Method | Composer

#### Summary

Regression of https://github.com/sebastianbergmann/phpunit/issues/3506

@remicollet can you confirm?

**Comment by robsontenorio** [comment]
For future reference


Not work as expected
```
php -d pcov.enabled=1 ./vendor/bin/phpunit 
```

WORKS!
```bash
./vendor/bin/phpunit -d pcov.enabled=1
```

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by robsontenorio** [comment]
Hi @sebastianbergmann ! Thanks for answering :)

Here is the  minimal, self-contained, reproducing test case that shows my issue. README included.

https://github.com/robsontenorio/bug-phpunit

It seems php options are not passed to child proccess when using **`@runTestsInSeparateProcesses`**

php **-d pcov.enabled=1** ./vendor/bin/phpunit .....

**Comment by krakjoe** [comment]
```
        foreach ($files as $ini) {                       
            $config = parse_ini_file($ini, true);       
                              
            foreach ($values as $value) {          
                $set = ini_get($value);                              
                                                                     
                if ($set == null) {             
                        continue;                
                }                                           
                                                             
                if ((!isset($config[$value]) || ($set != $config[$value]))) {
                    $diff[$value] = sprintf('%s=%s', $value, $set);  
                }                      
            }                                                                    
        }  
```

Could try that ...

**Comment by ktomk** [comment]
@krakjoe:

Given $config is with sections (parse_ini_file(, true)) and $value is the name of the ini directive, $config[$value] never is the value of the directive of the same name as in $value.

$config = parse_ini_file($ini, true);   -> $config = parse_ini_file($ini, false);  

https://3v4l.org/TMrtr

**Comment by robsontenorio** [comment]
Please, note pcov is disabled by default. As suggested on its docs. 

I am enabling pcov on demand with “php -d pcov.enabled=1 ...”

**Comment by krakjoe** [comment]
I'm going to have a closer look ... parse_ini_file does not behave exactly the same as the ini parser internally. Internally there are no sections. But we can't just switch to parsing without sections without loosing zend_extension/extension directives. And we're not doing anything with those directives currently.

I'll come up with something better ...

**Comment by ciltocruz** [comment]
What about this? My tests doesn't recognize xdebug extension when coverage is executed and my test has `@runTestsInSeparateProcesses` annotation...

**Comment by ktomk** [comment]
> What about this? My tests doesn't recognize xdebug extension when coverage is executed and my test has `@runTestsInSeparateProcesses` annotation...
until further traction, please try with the XDEBUG_MODE environment parameter (exported), it works across all processes in the same environment (as long as the PHP CLI config loads the xdebug extension by default), so it won't need the mode ini setting passed along explicitly via command-line args (explicitly by `-d` flags or via a temporary ini-file)  @ciltocruz

**Comment by ciltocruz** [comment]
> > What about this? My tests doesn't recognize xdebug extension when coverage is executed and my test has `@runTestsInSeparateProcesses` annotation...
> 
> until further traction, please try with the XDEBUG_MODE environment parameter (exported), it works across all processes in the same environment (as long as the PHP CLI config loads the xdebug extension by default), so it won't need the mode ini setting passed along explicitly via command-line args (explicitly by `-d` flags or via a temporary ini-file) @ciltocruz

Thanks for your help, @ktomk.
This is only for Xdebug 3?

I am using PHPStorm on Windows... Maybe this could be a problem...

**Comment by ktomk** [comment]
> This is only for Xdebug 3?

Yes, the [XDEBUG_MODE environment parameter](https://xdebug.org/docs/all_settings#XDEBUG_CONFIG) is for the currently supported Xdebug version only and does not apply to Xdebug version 2.

> I am using PHPStorm on Windows... Maybe this could be a problem...

Please see [_Environment Variables_ - _Command Line area﻿_ - _PHPUnit﻿_ ](https://www.jetbrains.com/help/phpstorm/run-debug-configuration-phpunit.html#commandLine) ... maybe it helps.

**Comment by ciltocruz** [comment]
Hello, @ktomk 

With the environment variable, my tests works.

Thanks for your help.

**Comment by dingo-d** [comment]
Any words on fixing this issue?

**Comment by sebastianbergmann** [comment]
I do not have time to work on this myself, but I would take the time to review a pull request.

**Comment by sebastianbergmann** [comment]
@robsontenorio Can you verify that https://github.com/sebastianbergmann/environment/pull/63 fixes this for you? Thanks!

**Comment by dingo-d** [comment]
@sebastianbergmann I've tested the issue @robsontenorio reported, when I put the PHPUnit to version ^8 I got

```bash
bash-5.1# php -d pcov.enabled=1 ./vendor/bin/phpunit
PHPUnit 8.5.26 #StandWithUkraine

  Warning - The configuration file did not pass validation!
  The following problems have been detected:

  Line 11:
  - Element 'coverage': This element is not expected.

  Test results may not be as expected.


..                                                                  2 / 2 (100%)

Time: 232 ms, Memory: 4.00 MB
```

When I set the PHPUnit to be ^9.5, the latest environment package gets pulled and the results are correct:

```bash
bash-5.1# php -d pcov.enabled=1 ./vendor/bin/phpunit
PHPUnit 9.5.20 #StandWithUkraine

..                                                                  2 / 2 (100%)

Time: 00:00.357, Memory: 10.00 MB

OK (2 tests, 2 assertions)

Generating code coverage report in Clover XML format ... done [00:00.018]

Generating code coverage report in HTML format ... done [00:00.156]
```

So I'd say that the issue is fixed.
